tests/README.md

```diff
@@ -0,0 +1,1510 @@
+
+```php
+<?php
+	/**
+	 * @Test
+	 * @small
+	 */
+	public function testCreateRelativePathWithWebTargetWhichShouldSucceed() {
+		$expectedPath = 'testtarget/subdir';
+		$target = $this->webPath . '/' . $expectedPath;
+
+		$result = LegacyPathHelper::createRelativePath($target);
+
+		$this->assertEquals($expectedPath, $result);
+	}
+```
+
+<a name="good-tests-are-first"></a>
+### Good tests are FIRST
+Many problems while unit testing can be avoided by following the FIRST principles (see Langr 2015).
+- **F** ast
+- **I** solated
+- **R** epeatable
+- **S** elf-validating
+- **T** imely
+
+<a name="fast"></a>
+#### Fast
+Keep the unit tests as fast as possible. They will be run multiple times a day to verify the
+behaviour of all the classes.
+
+The small grade between fast and slow test can be sometimes a bit blurred, however if the tested part
+of the code opens database connection or operates with files on a real filesystem the test are always slow.
+
+A lot of slow unit tests are usually an indicator of not so well designed component. Because all the code
+is tightly coupled to the slow parts or operation of a system.   
+
+<a name="isolated"></a>
+#### Isolated
+Unit-Test focus on small junks of code. So called single *unit*. The more code a test involves the
+more likely it is that the test fails out of unreasonably circumstances.
+
+For example the code which is under test might interact with other code which connects and interacts with
+a database. The database itself needs an entire host. So, in fact the test depends on that database and the
+structure as well as the data. If the datasource is shared between developers the result of the tests are
+no longer reliable because of external changes which are out of control for each individual developer.
+
+Good unit tests also don't depend on other unit tests. For example, all unit tests depend on each other to
+safe some time creating expensive objects. After some time something goes wrong, there will be a
+massive amount of time spend to found the actual cause because everything is failing
+due to the high coupling between each test.
+
+Therefore, unit test must be executable any time in any possible order.
+
+The Single Responsibility Principle (SRP) of the SOLID class design principle describes that class should be
+small and only serve one purpose. This principe is also really good for unit tests because if a test can for more than one
+reason. It's the best to split the test in multiple cases. If a focused test breaks it is normally obvious why.
+
+<a name="repeatable"></a>
+#### Repeatable
+A repeatable test is one which creates the same results all the time. In order to accomplish that, the test must be *isolated*.
+Each system will interact with elements which are not under the control of the developers. For example, if a system has to deal with
+dates or time. That means this test have to deal with additional problem which makes writing them more difficult.
```

Comment:
Yes, I totally agree with you. This would solve the problem, however I think its worth a warning as long as we don't have such an abstraction or would you like to remove it @klees ?

Reply:
Maybe you could just link to the "Mock Statics" chapter, because the problem is reiterated there, right?